dbt/adapters/duckdb/plugins/glue.py

```diff
@@ -172,18 +213,35 @@ def _add_partition_columns(table_def: TableInputTypeDef, partition_columns) -> T
         partition_keys.append(partition_column)
     table_def["PartitionKeys"] = partition_keys
     # Remove columns from StorageDescriptor if they match with partition columns to avoid duplicate columns
-    for partition_column in partition_columns:
+    for p_column in partition_columns:
         table_def["StorageDescriptor"]["Columns"] = [
             column
             for column in table_def["StorageDescriptor"]["Columns"]
-            if not (
-                column["Name"] == partition_column["Name"]
-                and column["Type"] == partition_column["Type"]
-            )
+            if not (column["Name"] == p_column["Name"] and column["Type"] == p_column["Type"])
         ]
     return table_def
 
 
+def _parse_partition_columns(
+    partition_columns: List[Dict[str, str]], table_def: TableInputTypeDef
+):
+    partition_input = None
+    if partition_columns:
+        partition_values = [column["Value"] for column in partition_columns]
+        partition_location = table_def["StorageDescriptor"]["Location"]
+        partition_components = [partition_location]
+        for c in partition_columns:
+            partition_components.append("=".join((c["Name"], c["Value"])))
+        partition_location = "/".join(partition_components)
+
+        partition_input = PartitionInputTypeDef()
+        partition_input["Values"] = partition_values
+        partition_input["StorageDescriptor"] = table_def["StorageDescriptor"]
+        partition_input["StorageDescriptor"]["Location"] = partition_location
+
+    return partition_input, partition_values
```

Comment:
sorry, does this still work if `partition_columns` is empty? I figured it would throw an error since `partition_values` wouldn't be defined.

Reply:
I'm checking if partition_columns != [] before calling this function as it shouldn't be called with empty partition_colums. Do you prefer that check inside the _parse_partition_columns function?

Reply:
Oh I just meant that a default `None` for `partition_values` should be on the same line as the default value for `partition_input`